app/com/arpnetworking/kairos/client/models/MetricsQuery.java

```diff
@@ -0,0 +1,547 @@
+            /**
+             * Public constructor.
+             */
+            public Builder() {
+                super(Aggregator::new);
+            }
+
+            /**
+             * Sets the name of the aggregator. Required. Cannot be null or empty.
+             *
+             * @param value the name of the aggregator
+             * @return this {@link Builder}
+             */
+            public Builder setName(final String value) {
+                _name = value;
+                return this;
+            }
+
+            /**
+             * Sets the sampling of the aggregator. Optional. Cannot be null.
+             *
+             * @param value the sampling for the aggregator
+             * @return this {@link Builder}
+             */
+            public Builder setSampling(final Sampling value) {
+                _sampling = value;
+                return this;
+            }
+
+            /**
+             * Sets the sampling of the aggregator. Cannot be null.
+             *
+             * @param value the sampling for the aggregator
+             * @return this {@link Builder}
+             */
+            public Builder setArgs(final Map<String, Object> value) {
+                _args = value;
+                return this;
+            }
+
+            @NotNull
+            @NotEmpty
+            private String _name;
+            private boolean _alignSampling = true;
+            @NotNull
+            private Sampling _sampling = new Sampling.Builder().build();
+            @NotNull
+            private Map<String, Object> _args = Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Model class to represent the sampling field of an aggregator.
+     *
+     * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+     */
+    public static final class Sampling {
+        public String getUnit() {
+            return _unit;
+        }
+
+        public int getValue() {
+            return _value;
+        }
+
+        private Sampling(final Builder builder) {
+            _unit = builder._unit;
+            _value = builder._value;
+        }
+
+        @JsonProperty("unit")
+        private final String _unit;
+        @JsonProperty("value")
+        private final int _value;
+
+        /**
+         * Implementation of the builder pattern for a {@link Sampling}.
+         *
+         * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+         */
+        public static final class Builder extends OvalBuilder<Sampling> {
+            /**
+             * Public constructor.
+             */
+            public Builder() {
+                super(Sampling::new);
+            }
+
+            /**
+             * Sets the time unit. Required. Cannot be null or empty.
+             *
+             * @param value the time unit for the sampling
+             * @return this {@link Builder}
+             */
+            public Builder setUnit(final String value) {
```

Comment:
There should probably be an enumeration for this.

Reply:
Possibly.  I don't really want to mirror everything about Kairos' API, though.

Reply:
Isn't that more or less the point of the client view model? I understand if we are not using it, but in this case I assume we are using units.

Reply:
Fixed by not allowing unit to be set.